static/app/views/issueList/sidebar.tsx

```diff
@@ -38,22 +44,34 @@ class IssueListSidebar extends Component<Props, State> {
     onQueryChange: function () {},
   };
 
-  state: State = {
-    queryObj: queryToObj(this.props.query),
-    textFilter: queryToObj(this.props.query).__text,
-  };
+  state: State = this.parseQueryToState(this.props.query);
+
+  parseQueryToState(query: string): State {
+    const parsedResult: ParseResult = parseSearch(query) ?? [];
+    const textFilter = parsedResult
+      ?.filter(p => p.type === Token.FreeText)
+      .map(p => p.text)
+      .join(' ');
+    const parsedFilers = parsedResult?.filter(
+      p => p.type === Token.Filter
+    ) as TokenResult<Token.Filter>[];
```

Comment:
There's a better way to predicate with a filter: https://stackoverflow.com/a/59629419/790169

```tsx
const parsedFilers = parsedResult?.filter(
  (p): p is TokenResult<Token.Filter> => p.type === Token.Filter
);
```

Reply:
Thanks 👍